### ISSUE TITLE
Error while running ./ADtoLDAP

### ISSUE THREAD

**Issue by hxvxt2** [nohupped/ADtoLDAP#21]
Hi,

I am running into an issue when I trying to run `./ADtoLDAP`. I made sure the base schema of my AD is the same as the base schema of my LDAP instance.

```
main.go:104: DEBUG: Add : &{cn=Users,dc=myorg,dc=local [{objectClass [top container]} {cn [Users]}]}
main.go:107: ERROR: LDAP Result Code 21 "Invalid Attribute Syntax": objectClass: value #1 invalid per syntax
main.go:104: DEBUG: Add : &{cn=Computers,dc=myorg,dc=local [{objectClass [top container]} {cn [Computers]}]}
main.go:107: ERROR: LDAP Result Code 21 "Invalid Attribute Syntax": objectClass: value #1 invalid per syntax
main.go:104: DEBUG: Add : &{cn=System,dc=myorg,dc=local [{objectClass [top container]} {cn [System]}]}
main.go:107: ERROR: LDAP Result Code 21 "Invalid Attribute Syntax": objectClass: value #1 invalid per syntax
main.go:104: DEBUG: Add : &{cn=LostAndFound,dc=myorg,dc=local [{objectClass [top lostAndFound]} {cn [LostAndFound]}]}
main.go:107: ERROR: LDAP Result Code 21 "Invalid Attribute Syntax": objectClass: value #1 invalid per syntax
main.go:123: DEBUG: Done from func FindDels
main.go:104: DEBUG: Add : &{cn=Infrastructure,dc=myorg,dc=local [{objectClass [top infrastructureUpdate]} {cn [Infrastructure]}]}
main.go:107: ERROR: LDAP Result Code 21 "Invalid Attribute Syntax": objectClass: value #1 invalid per syntax
```

Could you please help me what could be wrong?

Thanks

**Comment by muzzol** [comment]
have you tried to add those objects by hand?
maybe with another frontend you'll get more info

**Comment by nohupped** [comment]
Could you share your `ini` file (with masked password) please?

**Comment by nohupped** [comment]
> have you tried to add those objects by hand?
> maybe with another frontend you'll get more info

Agree with this. I used phpLdapAdmin as my Ldap Frontend.

**Comment by nohupped** [comment]
`LDAP Result Code 21` is [defined](https://ldap.com/ldap-result-code-reference-core-ldapv3-result-codes/#rc-invalidAttributeSyntax) as
```
The invalidAttributeSyntax result code indicates that the requested add or modify operation would have resulted in an entry that had at least one attribute value that does not conform to the constraints of the associated attribute syntax.
```
The way I had to fix it in my use case was by adding relevant mapping of attributes under the `Map` section in the `ldapsync.ini` file. I hope you found a way to fix this, or is no more relevant to you. Closing this issue because there is no response.
